From b43ca0b519eaf87733fd760272003c60da193c98 Mon Sep 17 00:00:00 2001
From: Heinrich Schuchardt <xypron.glpk@gmx.de>
Date: Mon, 1 Oct 2018 09:51:07 +0200
Subject: [PATCH 1/1] eth: dm: fec: avoid CACHE alignment warning

If CONFIG_SYS_CACHELINE_SIZE is greater than ARCH_DMA_MINALIGN a warning
'CACHE: Misaligned operation at range [&, &]' may occur.

Call flush_dcache_range() with a valid range.

Signed-off-by: Heinrich Schuchardt <xypron.glpk@gmx.de>
---
 drivers/net/fec_mxc.c | 9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

diff --git a/drivers/net/fec_mxc.c b/drivers/net/fec_mxc.c
index dac07b6e34..2732efd65b 100644
--- a/drivers/net/fec_mxc.c
+++ b/drivers/net/fec_mxc.c
@@ -27,6 +27,10 @@
 
 #include "fec_mxc.h"
 
+#ifndef CONFIG_SYS_CACHELINE_SIZE
+#define CONFIG_SYS_CACHELINE_SIZE ARCH_DMA_MINALIGN
+#endif
+
 DECLARE_GLOBAL_DATA_PTR;
 
 /*
@@ -680,8 +684,9 @@ static int fec_send(struct eth_device *dev, void *packet, int length)
 #endif
 
 	addr = (ulong)packet;
-	end = roundup(addr + length, ARCH_DMA_MINALIGN);
-	addr &= ~(ARCH_DMA_MINALIGN - 1);
+	end = roundup(addr + length,
+		      max(ARCH_DMA_MINALIGN, CONFIG_SYS_CACHELINE_SIZE));
+	addr &= ~(max(ARCH_DMA_MINALIGN, CONFIG_SYS_CACHELINE_SIZE) - 1);
 	flush_dcache_range(addr, end);
 
 	writew(length, &fec->tbd_base[fec->tbd_index].data_length);
-- 
2.11.0

